Packs/FeedGithub/Integrations/FeedGithub/FeedGitHub.py

```diff
@@ -0,0 +1,603 @@
+    parser = plyara.Plyara()
+    raw_rules = parser.parse_string(text_content)
+    current_time = datetime.now().isoformat()
+    for parsed_rule in raw_rules:
+        try:
+            metadata = {key: value for d in parsed_rule["metadata"] for key, value in d.items()}
+            value_ = parsed_rule["rule_name"]
+            type_ = "YARA Rule"
+            mapper = {
+                "value": value_,
+                "description": metadata.get("description", ""),
+                "author": metadata.get("author", ""),
+                "rulereference": metadata.get("reference", ""),
+                "sourcetimestamp": metadata.get("date", ""),
+                "ruleid": metadata.get("id", ""),
+                "rulestrings": make_grid_layout(parsed_rule.get("strings", {})),
+                "condition": " ".join(parsed_rule["condition_terms"]),
+                "references": file_path,
+                "rawrule": f"```\n {plyara.utils.rebuild_yara_rule(parsed_rule)} \n```",
+            }
+            indicator_obj = {
+                "value": value_,
+                "type": type_,
+                "service": "github",
+                "fields": mapper,
+                "score": Common.DBotScore.NONE,
+                "firstseenbysource": current_time,
+                "rawJSON": {"value": value_, "type": type_},
+            }
+            parsed_rules.append(indicator_obj)
+        except Exception as e:
+            demisto.error(f"Rull: {parsed_rule} cannot be processed. Error Message: {e}")
+            continue
+    return parsed_rules
+
+
+def make_grid_layout(list_dict):
+    return [
+        {"index": d.get("name"), "string": d.get("value"), "type": d.get("type"), "modifiers": d.get("modifiers", "")}
+        for d in list_dict
+    ]
+
+
+def get_yara_indicators(content: list[dict]):
+    """
+    Retrieves YARA indicators from a list of content items.
+
+    Args:
+        content (list): A list of strings containing YARA rules.
+
+    Returns:
+        list: A list of dictionaries representing parsed and mapped YARA rules for each content item.
+    """
+    return [rule for item in content for rule in parse_and_map_yara_content(item)]
+
+
+def detect_domain_type(domain: str):
+    """
+    Detects the type of an indicator (e.g., Domain, DomainGlob) using tldextract library.
+
+    Args:
+        domain (str): The indicator value to be analyzed.
+
+    Returns:
+        Optional[FeedIndicatorType]: The type of the indicator, or None if detection fails.
+    """
+    try:
+        no_cache_extract = tldextract.TLDExtract(cache_dir=False, suffix_list_urls=None)  # type: ignore
+
+        if no_cache_extract(domain).suffix:
+            if "*" in domain:
+                return FeedIndicatorType.DomainGlob
+            return FeedIndicatorType.Domain
+
+    except Exception:
+        demisto.debug(f"tldextract failed to detect indicator type. indicator value: {domain}")
+    return None
+
+
+ipv4Regex = (
```

Comment:
do these regexes still need to be here? Why cant we import from CSP

Reply:
These three regaxes that are in the CSP do not capture ip addresses well, so this is what Dror said we would do